server/test/unit/com/thoughtworks/go/server/materials/SCMMaterialSourceTest.java

```diff
@@ -150,6 +151,32 @@ public void shouldReloadSchedulableMaterialsOnConfigChange() {
     }
 
     @Test
+    public void shouldReloadSchedulableMaterialsOnConfigRepoChange() {
+        Set<MaterialConfig> schedulableMaterialConfigs = new HashSet<>(Arrays.asList(svnMaterial.config()));
+
+        when(goConfigService.getSchedulableSCMMaterials()).thenReturn(schedulableMaterialConfigs);
+        when(materialConfigConverter.toMaterials(schedulableMaterialConfigs)).thenReturn(new HashSet<>(Arrays.asList(svnMaterial)));
+
+        Set<Material> materials = source.materialsForUpdate();
+
+        assertThat(materials.size(), is(1));
+        assertTrue(materials.contains(svnMaterial));
```

Comment:
This isn't required.  Its been tested in another test. Since you are mocking the same thing after this, its not necessary to check if schedulable materials contained something before. https://github.com/gocd/gocd/pull/3809/files#diff-4844feeb7ab629be0b9f429e848090dbR166 ensures that there had been another shedulable material present.